### ISSUE TITLE
mscvrt not found when calling fits.open() on Python 3.5 on Win10

### ISSUE THREAD

**Issue by and-colonna** [astropy/astropy#4342]
When using `fits.open(...)` on Python 3.5 / Windows 10, I got error message:

```
C:\Python35\lib\site-packages\astropy\io\fits\util.py:577: UserWarning: Could not find appropriate MS Visual C Runtime library or library is corrupt/misconfigured; cannot determine whether your file object was opened in append mode.  Please consider using a file object opened in write mode instead.
  'Could not find appropriate MS Visual C Runtime '
```

I think pyreadline/pyreadline#32 (or parent issue ipython/ipython#8316 - specifically the phrase "`ctypes.util.find_msvcrt()` could be broken") might be related.

* [ ] Once the issue is fixed, Python 3.5 should be un-marked as an allowed failure in appveyor.yml -- See #4450

**Comment by embray** [comment]
Indeed, it is the same issue (that is, the use of `ctypes.util.find_msvcrt`).  In the case of `astropy.io.fits` it isn't really a serious issue, but good to know about so thanks for pointing it out.

**Comment by eteq** [comment]
So is this something that can be fixed in astropy, or does it require a more up-to-date pyreadline?  (That is, should we close this and/or add it to the "known issues" list?)

**Comment by embray** [comment]
@eteq This has nothing to do with pyreadline--it's just that the same bug is there.  This is something that needs, presumably, to be fixed in Astropy (really it's an upstream issue in Python that needs to be worked around).

**Comment by embray** [comment]
Also, it occurs to me that this warning is coming up in contexts where it's really pretty irrelevant.  It is only relevant when opening the file with `mode='append'`, and passing in an open file handle.  The code here is trying to determine that the raw file handle was opened in append mode (or at least write, in which case we can emulate append).  In the case of just passing in a filename we don't care, and the user shouldn't be bothered with this.  But changing this will require yet another bit of refactoring of the `_File` class.

**Comment by astrofrog** [comment]
* [ ] Once the issue is fixed, Python 3.5 should be un-marked as an allowed failure in appveyor.yml

**Comment by anntzer** [comment]
You may want to check https://bugs.python.org/issue26727 and the linked issue.

**Comment by eteq** [comment]
@MSeifert04 astutely pointed out that #4792 fixes this (as evidences by the 3.5 tests now passing), so I'll close this.
